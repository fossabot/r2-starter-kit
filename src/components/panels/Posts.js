import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import List from 'components/elements/List';
import Loading from 'components/elements/Loading';

class Posts extends PureComponent {
  static propTypes = {
    // container state
    list: PropTypes.arrayOf(PropTypes.object),
    state: PropTypes.oneOf(['loading', 'idle', 'failure']),

    // container actions
    getList: PropTypes.func,
  };
  static defaultProps = {
    // container state
    list: [],
    state: 'idle',

    // container actions
    getList: () => {},
  };
  componentDidMount() {
    this.props.getList();
  }
  render() {
    const { list, state } = this.props;
    return (
      <div>
        {state === 'loading' && <Loading />}
        {state === 'idle' && (
          <div className="ts text container">
            <List data={list} />
          </div>
        )}
      </div>
    );
  }
}

export default Posts;
