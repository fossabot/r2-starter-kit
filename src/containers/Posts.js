import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { getList } from 'redux/modules/list';
import Posts from 'components/panels/Posts';

function mapStateToProps(state) {
  return {
    list: state.list,
    state: state.state,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ getList }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Posts);
