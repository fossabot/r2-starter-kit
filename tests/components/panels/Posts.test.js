import React from 'react';
import { shallow } from 'enzyme';

import Posts from 'components/panels/Posts';

describe('<Posts />', () => {
  const initProps = {
    list: [],
    state: 'loading',

    getList: jest.fn(),
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Posts {...initProps} />);
  });

  it('should call getPosts in componentDidMount', () => {
    expect(initProps.getList).toHaveBeenCalledTimes(1);
  });

  it('should render loading page', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render post list page', () => {
    wrapper.setProps({
      list: [{
        title: 'Posts Title',
        author: {
          name: 'Tony Pai',
          email: 'tonypai@me.com',
        },
        body: 'Posts Content',
      }],
      state: 'idle',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
