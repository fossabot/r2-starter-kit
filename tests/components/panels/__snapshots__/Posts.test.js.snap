// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should render loading page 1`] = `
<div>
  <Loading />
</div>
`;

exports[`<Posts /> should render post list page 1`] = `
<div>
  <div
    className="ts text container"
  >
    <List
      data={
        Array [
          Object {
            "author": Object {
              "email": "tonypai@me.com",
              "name": "Tony Pai",
            },
            "body": "Posts Content",
            "title": "Posts Title",
          },
        ]
      }
    />
  </div>
</div>
`;
